primitives/linux-port/dictionary.cpp

```diff
@@ -597,7 +597,8 @@ void PrimitiveProcessor::p_Dictionary(const DictInput* in, vector<uint8_t>* out,
         outValue = reinterpret_cast<DataValue*>(&(*out)[header.NBYTES]);
         outValue->isnull = sigptr.data == nullptr;
         outValue->len = sigptr.len;
-        memcpy(outValue->data, sigptr.data, sigptr.len);
+        if (sigptr.data != nullptr)
```

Comment:
IMHO, if we compare sigptr.data with null ptr the else should log this.
We could replace this if with idbassert but it doesn't look good to assert so logging suits better IMHO.